src/http/server/response.cr

```diff
@@ -126,22 +126,32 @@ class HTTP::Server
     # Sends an error response.
     #
     # Calls `#reset`, writes the given status, and closes the response.
-    def respond_with_error(status : HTTP::Status)
-      respond_with_error(status.description, status.code)
+    @[Deprecated("Use #respond_with_status instead")]
+    def respond_with_error(message = "Internal Server Error", code = 500)
+      respond_with_status(code, message)
     end
 
-    # Sends an error response.
+    # Sends a *status* and *message* as response.
     #
-    # Calls `#reset`, writes the given message, and closes the response.
-    def respond_with_error(message = "Internal Server Error", code = 500)
+    # This method calls `#reset` to remove any previous settings and writes the
+    # given *status* and *message* to the response IO. Finally, it closes the
+    # response.
+    #
+    # If *message* is `nil`, the default message for *status* is used provided
+    # by `HTTP::Status#description`.
+    def respond_with_status(@status : HTTP::Status, message : String? = nil)
       reset
-      @status = HTTP::Status.new(code)
       message ||= @status.description
       self.content_type = "text/plain"
       self << @status.code << ' ' << message << '\n'
       close
     end
 
+    # :ditto:
+    def respond_with_status(status : Int = 500, message : String? = nil)
```

Comment:
*status* argument IMO should be required.

Reply:
Yeah, I guess that's better. Before, it was `respond_with_error` where one could argue that a default error status is 500. But even that's not necessarily the case. So, yeah, I'll make it required.